ISISComputingGroup/EPICS-ioc
# Add actions for post-begin/post-end for SANS2D fast shutter

### Description of work

Adds post-begin/post-end hooks for the ISISDAE, and use them to control SANS2D's fast shutter.

### To test

https://github.com/ISISComputingGroup/IBEX/issues/4590

Add the following to your globals.txt (replacing `TE:NDW1799:` with your pv prefix):

```
ISISDAE_01__POST_BEGIN_1=TE:NDW1799:FINS_VAC:SHUTTER:OPEN_IF_AUTO.PROC CA PP
ISISDAE_01__POST_RESUME_1=TE:NDW1799:FINS_VAC:SHUTTER:OPEN_IF_AUTO.PROC CA PP
ISISDAE_01__POST_END_1=TE:NDW1799:FINS_VAC:SHUTTER:CLOSE_IF_AUTO.PROC CA PP
ISISDAE_01__POST_ABORT_1=TE:NDW1799:FINS_VAC:SHUTTER:CLOSE_IF_AUTO.PROC CA PP
ISISDAE_01__POST_PAUSE_1=TE:NDW1799:FINS_VAC:SHUTTER:CLOSE_IF_AUTO.PROC CA PP
```

ensure you have rebuilt in `ioc/fins` and `support/isisdae/master`

### Acceptance criteria

If the shutter control is in manual mode:
- beginning runs from the GUI or genie_python does not change the shutter state
- ending runs from the GUI or genie_python does not change the shutter state

If the shutter control is in automatic mode:
- beginning runs from the GUI or genie_python opens the fast shutter (test that `$(P)FINS_VAC:SHUTTER:STATUS:SP` gets changed to `OPEN`)
- ending runs from the GUI or genie_python closes the fast shutter  (test that `$(P)FINS_VAC:SHUTTER:STATUS:SP` gets changed to `CLOSE`)

---

#### Code Review

    - [PV naming](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/PV-Naming).
    - [Disable records](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Disable-records)
    - [Record simulation](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Record-Simulation)
    - [Finishing touches](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/IOC-Finishing-Touches)

### Functional Tests

- IOC responds correctly in:

### Final steps
